### PR TITLE
Refactor calc.lib.base

### PR DESCRIPTION
**Pull request by michele-riva**
Creates a few submodules of `calc.lib` that collect various related functions from `calc.lib.base`. New modules:
- `coordinates.py`: functions for handling atomic positions.
- `fortran_utils.py`: functionality for FORTRAN code.
- `itertools_utils.py`: backward-compatible `itertools` functions, plus some new ones.
- `log_utils.py`: `logging` stuff, including new context managers for silencing a logger or the whole `logging`.
- `math_utils.py`: useful mathematical functions.
- `matrix.py`: matrix-related exceptions and functions, and transformation matrices.
- `sequence_utils.py`: functions specifically running on sequences only (those for iterables are in `itertools_utils`).
- `string_utils.py`: functions operating on strings.

Also reimplements some of the moved functions for clarity/performance, and adds a few new functions.

Some of the functions still left in `base` could be moved to the new modules, but are very likely to be removed at a later stage as they're e bit funny and not very readable. Here a list of functions that I think should be removed:
- `dict_equal`: not very useful, considering that dictionaries support `==`. Used in `sitetype`.
- `parseMathSqrt`: should be replaced by the more general functionality in `guilib` for parsing mathematical expressions. Pending my own refactor.
- `dist_from_line`: pending #51.
- `readToExc`: used in `vibrocc`. That's the only file that currently has inconsistent handling of comment characters. I have a full reimplementation in my stash of code that will eventually become a PR.
- `splitSublists`: confusing. See my comment at 212502cc479e71936f9c7d454de2d7bbfe25734d.
- `addUnequalPoints`: will be removed in #51.

All new modules are 100% covered by tests.
All other tests passing.

### CODE REVIEW

File: src/viperleed/calc/lib/base.py [hunk 1]
```diff
@@ -597,41 +288,6 @@ def dist_from_line(p1, p2, r):
...
-    """Return the part of line to the left of comments."""
-    for comment_char in "!#%":
-        try:
-            line, *_ = line.split(comment_char)
-        except ValueError:  # Nothing left to split
-            return ''
-    return line.strip()
-
-
 def readToExc(llist):                                                           # TODO: unused; could be an iterator
```
Comment (amimre) [inline src/viperleed/calc/lib/base.py:291]: can it be removed?

**Review by amimre — APPROVED** [review]

**Comment by amimre** [comment]
Looks good to me!

**Comment by amimre** [comment]
Maybe some things from lib.base could move into a lib.math library (such as readVec, angle etc.)...

**Comment by michele-riva** [comment]
I'm now done with refactoring also the rest as we have discussed in person. I have not moved the matrices that are defined in `from_ase` to the `matrix` module on purpose, as they are currently only used there, and have some inconsistency of naming and signature with respect to the ones in `matrix`. Also, `from_ase` only defines rotations around 'x' and 'z', but not around 'y'. I feel these would look funny in a `lib` module.

**Comment by michele-riva** [comment]
Perhaps hold off with reviewing this till tomorrow morning, as I forgot to re-run the whole test suite. I'll comment again when I'm sure all tests indeed pass my second version.

**Comment by michele-riva** [comment]
Alright, I can now confirm that, after my recent edits, all tests do pass. @amimre, may I ask you to review this again?

**Review by amimre — APPROVED** [review]

**Pull request merged**